### PR TITLE
Update to Userspace kuchosauronad0

### PR DESCRIPTION
**Pull request by kuchosauronad0**
* Updated my userspace with RGB
* Added the plover layer back
* Leaderkey now has an indicator for success/failed sequences.
* Cleaned up a bit.

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [ ] Keyboard (addition or update)
- [x] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Checklist

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: users/kuchosauronad0/rgblight_user.c [hunk 1]
```diff
@@ -0,0 +1,157 @@
...
+
+#ifdef INDICATOR_LIGHTS
+void matrix_scan_indicator(void) {
+  if (has_initialized) {
+    set_rgb_indicators(get_mods(), host_keyboard_leds(), get_oneshot_mods());
+  }
+}
+#endif // !INDICATOR_LIGHTS
+
+void rgblight_fade_helper(bool direction){
```
Comment (drashna) [inline users/kuchosauronad0/rgblight_user.c]: Very cool!

File: users/kuchosauronad0/rgblight_user.c [hunk 2]
```diff
@@ -0,0 +1,157 @@
...
+  if (userspace_config.rgb_layer_change) {
+    rgblight_enable_noeeprom();
+  }
+  if (rgblight_config.enable) {
+    layer_state_set_user(layer_state);
+    uint16_t old_hue = rgblight_config.hue;
+    rgblight_mode_noeeprom(RGBLIGHT_MODE_STATIC_LIGHT);
+    for (uint16_t i = 255; i > 0; i--) {
+      rgblight_sethsv_noeeprom((i + old_hue) % 255, 255, 255);
+      matrix_scan();
```
Comment (zvecr) [inline users/kuchosauronad0/rgblight_user.c]: Maybe I am missing context, but what does calling matrix_scan do here?

File: users/kuchosauronad0/kuchosauronad0.h [hunk 3]
```diff
@@ -1,8 +1,7 @@
+#pragma once
 #ifndef USERSPACE
```
Comment (zvecr) [inline users/kuchosauronad0/kuchosauronad0.h]: Unless your are checking this is defined somewhere, `#pragma once` replaces these guards
```suggestion
```

File: users/kuchosauronad0/kuchosauronad0.h [hunk 4]
```diff
@@ -1,8 +1,7 @@
...
 #define USERSPACE
```
Comment (zvecr) [inline users/kuchosauronad0/kuchosauronad0.h]:
```suggestion
```

File: users/kuchosauronad0/kuchosauronad0.h [hunk 5]
```diff
@@ -95,17 +97,18 @@ But since TD() doesn't work when tap dance is disabled
...
+#  define KC_D3_3 TD(TD_D3_3)
+#  define KC_D3_4 TD(TD_D3_4)
+#else
+#  define KC_D3_1 KC_1
+#  define KC_D3_2 KC_2
+#  define KC_D3_3 KC_3
+#  define KC_D3_4 KC_4
+#endif // !TAP_DANCE_ENABLE
 
 #endif // !USERSPACE
```
Comment (zvecr) [inline users/kuchosauronad0/kuchosauronad0.h]:
```suggestion
```

File: users/kuchosauronad0/rgblight_user.c (hunk 2 above)
Comment (drashna) [reply users/kuchosauronad0/rgblight_user.c]: I have matrix scan there, so that it will run the rgb matrix animations, rather than halt them.

**Pull request merged**